### DISCUSSION TITLE
[Compiled VS Code Linux] Copilot claims vscode is outdated, yet running latest release

### DISCUSSION THREAD

**Discussion by Kaydax** [community/community#6629] [Copilot Conversations] [answered]
I'm trying to use copilot on visual studio oss 1.61.2, yet copilot says that the current vscode version is outdated.

Visual Studio Code is compiled from source, from the commit tag 1.61.2:
![image](https://user-images.githubusercontent.com/11893096/138530299-6bcfae42-895d-4b6d-9b75-f16cfc1351f9.png)

**Comment by sheepy0125** [comment]
I'm having this issue too! And on VSCodium, it just silently fails...

**Reply by davecheney** [reply]
Hello!

It's likely that internal build differences between VSCodium and VSCode are breaking Copilot's VSCode version check. Ideally we should support VSCodium, and I'm sure we will in the future, but for now, would you please use VSCode.

Thanks

**Reply by Kaydax** [reply]
I am using VSCode and still experience this issue

**Reply by davecheney** [reply]
I'm sorry you're having issues. I'm not sure what to suggest other than the usual uninstall/reinstall/turn it off/on again sort of thing.

**Reply by Kaydax** [reply]
Tried that multiple times now with no success. Im compiling vscode from source and yet it just doesnt seem to think im on the right version

**Reply by davecheney** [reply]
Possibly it's caused by building from source. I've never tried that so I can't really comment on what the differences are, but I believe if you use the version of VSCode from https://code.visualstudio.com/ it will work for you. I understand if you choose not to do that. As I said, we'll fix up the version detection code to be more lenient in the future.

**Reply by sheepy0125** [reply]
> Hello!
> 
> It's likely that internal build differences between VSCodium and VSCode are breaking Copilot's VSCode version check. Ideally we should support VSCodium, and I'm sure we will in the future, but for now, would you please use VSCode.
> 
> Thanks

I should clarify: I've tried both Code - OSS and VSCodium. On Code - OSS, it gives the version check error, and on VSCodium, it doesn't push any notifications at all.

But, on Windows with VSCode, it works perfectly, so I'll be using that for now :)

**Reply by Kaydax** [reply]
Code - OSS is the linux version of vscode as well, as in its directly from both the site and package managers alike

**Reply by jcuenod** [reply]
I'm having the same issue on Code (Arch). I suppose that [`visual-studio-code-bin`](https://aur.archlinux.org/packages/visual-studio-code-bin/) is the solution... For a list of options on Arch, see https://wiki.archlinux.org/title/Visual_Studio_Code#Installation

**Reply by fxtressia** [reply]
> Code - OSS is the linux version of vscode as well, as in its directly from both the site and package managers alike

Actually, it isn't. Proprieritary VSCode is also available for Linux, but software distributions generally recommends the open source build of it as it's open source.

**Comment by Krzysztof-Cieslak** [comment, marked as answer]
Hey, 

GitHub Copilot is using some of the APIs that are still marked as proposed in the VSCode. They need to be manually enabled on non-standard VSCode releases. 
To enable those APIs, in your version of VSCode, run `Preferences: Configure Runtime Arguments` command, and in the JSON file add following entry: `"enable-proposed-api":["github.copilot"]`. And then restart VSCode.

**Reply by Kaydax** [reply]
Thanks this seems to have done the trick. I was looking into that but I was confused as to where to enable it, as in terminal, launch script or other places, but doing it in a json file does the trick. It would be nice if the error said it was needed instead of saying it was outdated

**Reply by jcuenod** [reply]
Thanks, this worked for me too

**Reply by sheepy0125** [reply]
Thank you!

**Reply by fxtressia** [reply]
For Copilot Nightly users, use "github.copilot-nightly" instead.

**Reply by Zauberbutter** [reply]
Since last update this no longer works for me. Even when I start it like that: `code --enable-proposed-api GitHub.copilot`. :neutral_face:

**Reply by Rigel772** [reply]
This does not work anymore....
```"enable-proposed-api": ["github.copilot"]}```

**Reply by Vanclief** [reply]
Having same issue, this doesn't work for me either anymore

**Reply by Rigel772** [reply]
What I found out is it does not work on VSCodium and OSS Code but works on VSCode when you install as described here: https://code.visualstudio.com/docs/setup/linux

**Reply by Krzysztof-Cieslak** [reply]
Hey, can y'all make sure you're running the latest version of Code (1.63.X)

**Reply by Zauberbutter** [reply]
@Krzysztof-Cieslak Still getting these errors: 

```
editor/inlineCompletions/actions is a proposed menu identifier. It requires 'package.json#enabledApiProposals: ["inlineCompletions"]' and is only available when running out of dev or with the following command line switch: --enable-proposed-api GitHub.copilot
'configuration.icons is a proposed contribution point. It requires 'package.json#enabledApiProposals: ["contribIcons"]' and is only available when running out of dev or with the following command line switch: --enable-proposed-api GitHub.copilot
'configuration.iconFonts is a proposed contribution point. It requires 'package.json#enabledApiProposals: ["contribIconFonts"]' and is and only available when running out of dev or with the following command line switch: --enable-proposed-api GitHub.copilot
```

```
code -v

1.63.2
899d46d82c4c95423fb7e10e68eba52050e30ba3
x64
```

with:

```
"enable-proposed-api": ["GitHub.copilot"]
```
